src/memory/query_memory_control.cpp

```diff
@@ -154,6 +154,12 @@ inline int &Get_Thread_Tracker() {
 
 bool QueriesMemoryControl::IsThreadTracked() { return Get_Thread_Tracker() == 1; }
 
+void QueriesMemoryControl::ResetTrackings() {
+  Get_Thread_Tracker() = 0;
+  transaction_id_to_tracker.clear();
+  thread_id_to_transaction_id.clear();
```

Comment:
I don't get it. If our GC run is exclusive (ie. no active transactions) we get to run this Reset. But in that situation why hasn't transaction end already unregistered itself. Ideally these collections should already be clear.

Reply:
Yes, they are clear, but this is just to have GC to be 100% sure.